tools/idf_tools.py

```diff
@@ -432,8 +432,7 @@ def download(url, destination):  # type: (str, str) -> Optional[Exception]
         # This works around the issue with outdated certificate stores in some installations.
         if 'dl.espressif.com' in url or 'github.com' in url:
             try:
-                ctx = ssl.create_default_context()
-                ctx.load_verify_locations(cadata=DIGICERT_ROOT_CERT)
+                ctx = ssl.create_default_context(purpose=Purpose.SERVER_AUTH, cadata=DIGICERT_ROOT_CERT)
```

Comment:
I think we should also call `ssl.load_default_certs` in this case, otherwise we will only load the DIGICERT_ROOT_CERT.

This is necessary since IDF_MIRROR_PREFIX_MAP and IDF_GITHUB_ASSETS environment variables can be used to redirect the download URLs to a mirror, which may be using a different certificate.

Reply:
Please check

Reply:
I'm not a Python expert so I can't comment further... 😅 
I do know that this change made the ANSI TAG error go away.
Maybe that's because of the cdata being passed here instead of `load_verify_locations`...
Feel free to test and change!